### PR TITLE
Update cmake to 3.27 and upgrade Linux CUDA docker files from CentOS7 to UBI8

### PR DESCRIPTION
**Pull request by snnn**
### Description
Update docker files and their build instructions

[AB#18990](https://aiinfra.visualstudio.com/6a833879-cd9b-44a4-a9de-adc2d818f13c/_workitems/edit/18990)
### Motivation and Context
ARM64 and x86_64 can use the same docker file.

### CODE REVIEW

**Review by jchen351 — DISMISSED** [review]

File: dockerfiles/scripts/install_common_deps.sh [hunk 1]
```diff
@@ -21,6 +21,6 @@ pip install "wheel>=0.35.1"
 rm -rf /opt/miniconda/pkgs
 
 # Dependencies: cmake
-wget --quiet https://github.com/Kitware/CMake/releases/download/v3.26.3/cmake-3.26.3-linux-x86_64.tar.gz
-tar zxf cmake-3.26.3-linux-x86_64.tar.gz
```
Comment (mszhanyi) [inline dockerfiles/scripts/install_common_deps.sh:25]: There's timeout issue in 3.27.0.
Does 3.27.1 fix it?

Comment (mszhanyi) [reply dockerfiles/scripts/install_common_deps.sh:25]: It looks the regression isn't fixed.
https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=334111&view=logs&jobId=f9f5b320-fa10-56c4-debe-61ea69c74793&j=f9f5b320-fa10-56c4-debe-61ea69c74793&t=73689a31-9570-57c6-acc6-0f29b528d037

**Review by mszhanyi — CHANGES_REQUESTED** [review]

File: dockerfiles/scripts/install_common_deps.sh (hunk 1 above)
Comment (mszhanyi) [reply dockerfiles/scripts/install_common_deps.sh:25]: It looks the regression isn't fixed.
https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=334111&view=logs&jobId=f9f5b320-fa10-56c4-debe-61ea69c74793&j=f9f5b320-fa10-56c4-debe-61ea69c74793&t=73689a31-9570-57c6-acc6-0f29b528d037

**Review by mszhanyi — CHANGES_REQUESTED** [review]

File: dockerfiles/scripts/install_cmake.sh [hunk 2]
```diff
@@ -3,5 +3,5 @@ mkdir -p /tmp/src
 cd /tmp/src
 
 echo "Installing cmake"
-curl https://github.com/Kitware/CMake/releases/download/v3.27.1/cmake-3.27.1-linux-`uname -m`.tar.gz -sSL --retry 5  -o /tmp/src/cmake-3.27.1-linux-`uname -m`.tar.gz
-tar -zxf /tmp/src/cmake-3.27.1-linux-`uname -m`.tar.gz --strip=1 -C /usr
+curl https://github.com/Kitware/CMake/releases/download/v3.27.2/cmake-3.27.2-linux-`uname -m`.tar.gz -sSL --retry 5  -o /tmp/src/cmake-3.27.2-linux-`uname -m`.tar.gz
```
Comment (thiagocrepaldi) [inline dockerfiles/scripts/install_cmake.sh:6]:
```suggestion
CMAKE_VERSION="3.27.2"
curl https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-linux-`uname -m`.tar.gz -sSL --retry 5  -o /tmp/src/cmake-${CMAKE_VERSION}-linux-`uname -m`.tar.gz
```

Comment (thiagocrepaldi) [reply dockerfiles/scripts/install_cmake.sh:6]: Declaring a bash variable to hold the version and use it everywhere will save a lot of copy&paste in the future :)

File: dockerfiles/scripts/install_common_deps.sh (hunk 1 above)
Comment (snnn) [reply dockerfiles/scripts/install_common_deps.sh:25]: 3.27.2 still has the issue.

**Review by mszhanyi — APPROVED** [review]

File: dockerfiles/scripts/install_common_deps.sh (hunk 1 above)
Comment (snnn) [reply dockerfiles/scripts/install_common_deps.sh:25]: I added a timeout value to each test target, then it is fine.

File: dockerfiles/scripts/install_cmake.sh (hunk 2 above)
Comment (snnn) [reply dockerfiles/scripts/install_cmake.sh]: Thanks for your great suggestion. I will do it in the next PR. 
I need to merge this PR first to bring nightly pipelines back to normal.

File: tools/android_custom_build/Dockerfile [hunk 3]
```diff
@@ -24,7 +24,7 @@ RUN apt-get update && apt-get install --yes --no-install-recommends \
   unzip lsb-release
 
 # cmake
-RUN CMAKE_VERSION=3.26.3 && \
+RUN CMAKE_VERSION=3.27.3 && \
   aria2c -q -d /tmp -o cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz \
   --checksum=sha-256=28d4d1d0db94b47d8dfd4f7dec969a3c747304f4a28ddd6fd340f553f2384dc2 \
```
Comment (edgchen1) [inline tools/android_custom_build/Dockerfile:29]: checksum needs to be updated

**Pull request merged**